src/API.jl

```diff
@@ -1649,9 +1649,19 @@ function _activate_dep(dep_name::AbstractString)
         end
     end
 end
-function activate(path::AbstractString; shared::Bool=false, temp::Bool=false, io::IO=stderr_f())
+function activate(path::AbstractString; shared::Bool=false, temp::Bool=false, io::IO=stderr_f(), show_shared_hint::Bool=true)
     temp && pkgerror("Can not give `path` argument when creating a temporary environment")
     if !shared
+        if show_shared_hint && startswith(path, "@")
+            r = r"^@(.*)$"
+            m = match(r, path)
+            name = m[1]
```

Comment:
```suggestion
            name = path[begin+1:end]
```

or `path[2:end]` as in https://github.com/JuliaLang/Pkg.jl/blob/02b6f7798bb4263d1fd7255ceeeef24403def650/src/REPLMode/argument_parsers.jl#L203